client/services/FeedbackGeneratorService.js

```diff
@@ -30,6 +30,65 @@ tie.factory('FeedbackGeneratorService', [
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
     PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR) {
+
+    /**
+     * Global constant to act as the threshhold before prompting the user
+     * to look at the language primer.
+     *
+     * @type {number}
+     */
+    var UNFAMILIARITY_THRESHOLD = 5;
+
+    /**
+     * Global number to keep track of consecutive syntax errors.
+     *
+     * @type {number}
+     */
+    var consecutiveSyntaxErrorCounter = 0;
+
+    /**
+     * Global number to keep track of consecutive same errors.
+     *
+     * @type {number}
+     */
+    var consecutiveSameErrorCounter = 0;
+
+    /**
+     * Global number to keep track of consecutive uses of the wrong language.
+     *
+     * @type {number}
+     */
+    var consecutiveWrongLanguageCounter = 0;
```

Comment:
consecutiveWrongLanguageErrorCounter? Otherwise you seem to be counting languages.